package/MDAnalysis/coordinates/GSD.py

```diff
@@ -0,0 +1,133 @@
+============================================================
+
+Classes
+-------
+
+.. autoclass:: GSDReader
+   :inherited-members:
+
+"""
+from __future__ import absolute_import, division
+
+import numpy as np
+from . import base
+try :
+    import gsd.hoomd
+except :
+    pass
+
+class GSDReader(base.ReaderBase):
+    """Reader for the GSD format.
+
+    TODO: write docstring
+
+    """
+    format = 'GSD'
+    units = {'time': None, 'length': None}
+    _Timestep = base.Timestep
+
+    def __init__(self, filename, **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str
+            trajectory filename
+        **kwargs : dict
+            General reader arguments.
+
+
+        .. versionadded:: 0.17.0
+        """
+        super(GSDReader, self).__init__(filename, **kwargs)
+        self.filename = filename
+        self.open_trajectory()
+        self.ts = self._read_next_timestep()
+
+    def open_trajectory (self) :
+        """opens the trajectory file using gsd.hoomd module"""
+        self._frame = -1
+        self._file = gsd.hoomd.open (self.filename,mode='rb')
+
+    def close(self):
+        """close reader"""
+        self._file.file.close()
+
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        return len(self._file)
+
+    def _reopen(self):
+        """reopen trajectory"""
+        self.close()
+        self.open_trajectory()
+
+    def _read_next_timestep(self):
+        """read next frame in trajectory"""
+        if self._frame == self.n_frames - 1:
+            raise IOError('trying to go over trajectory limit')
+        self._frame += 1
+        self.n_atoms = self._file[self._frame].particles.N
+        # instantiate the Timestep
+        frame = self._file[self._frame]
+        return self._frame_to_ts (frame)
+
+    def _frame_to_ts(self, frame):
+        """convert a gsd-frame to a :class:`TimeStep`"""
+        ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        ts.frame = frame.configuration.step
+
+        # set frame box dimensions
+        ts.dimensions = frame.configuration.box
+        for i in range(3,6) :
+            ts.dimensions[i] = np.arccos(ts.dimensions[i]) * 180.0 / np.pi
+
+        # set particle positions
+        frame_positions = frame.particles.position
+        n_atoms_now = frame_positions.shape[0]
+        if n_atoms_now != self.n_atoms :
+            raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                " atoms. MDAnalysis in unable to deal with variable"
+                " topology!"%(self._frame, n_atoms_now, self.n_atoms))
+        else :
+            ts.positions = frame_positions
+
+        self.ts = ts
+        return ts
+
+    @property
+    def dimensions(self):
+        """unitcell dimensions (*A*, *B*, *C*, *alpha*, *beta*, *gamma*)
+        """
+        return self.ts.dimensions
+
+    def __getitem__ (self,frame_numbers) :
```

Comment:
You shouldn't have to override this, is there a reason the base version isn't working?

Reply:
I did this because I got an error when attempting to access a trajectory slice. What's the correct way of doing it?

Reply:
For a Reader you should only have to implement `_read_frame(frame)` and `_read_next_timestep()`. The base class takes care of handling `__getitem__` etc [here](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/coordinates/base.py#L1280), it takes care of stuff like boolean slicing etc and makes all the Readers behave the same way.